botocore/endpoint_provider.py

```diff
@@ -236,8 +237,7 @@ def aws_parse_arn(self, value):
         arn_dict["accountId"] = arn_dict.pop("account")
 
         resource = arn_dict.pop("resource")
-        delimiter = ":" if ":" in resource else "/"
-        arn_dict["resourceId"] = resource.split(delimiter)
+        arn_dict["resourceId"] = re.split(ARN_DELIMITER_RE, resource)
```

Comment:
I'm not sure if there is a difference, but why not use split on the compiled object?

```suggestion
        arn_dict["resourceId"] = ARN_DELIMITER_RE.split(resource)
```